### PR TITLE
Migrate `checks` to GitHub Actions

### PR DESCRIPTION
**Pull request by crcrpar**

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install
+      run: |
+        python -m pip install -U pip
+        pip install --progress-bar off -U .[checking]
+
+    - run: black . --check
```
Comment (himkt) [inline .github/workflows/checks.yml:25]: How about adding option `--diff`?
related: https://github.com/optuna/optuna/pull/1840

**Review by hvy — COMMENTED** [review]
Thanks, it basically LGTM. This is just an idea on the cosmetic but how about naming the steps for flake8, isort, etc.?

**Review by hvy — APPROVED** [review]

File: .github/workflows/checks.yml (hunk 1 above)
Comment (crcrpar) [reply .github/workflows/checks.yml]: Thanks, done. https://github.com/optuna/optuna/pull/1838/commits/fb2754492af3351ce69987d437ff88fccde601b6#diff-521971030c28a8f612ca17754898dc4fR26

**Review by himkt — APPROVED** [review]
Thank you for the quick updates! LGTM.

**Pull request merged**